src/main/scala/com/codacy/rules/commituuid/providers/GitHubActionProvider.scala

```diff
@@ -1,15 +1,51 @@
 package com.codacy.rules.commituuid.providers
 
+import java.nio.file.Files
+import java.io.File
+
 import com.codacy.rules.commituuid.CommitUUIDProvider
 import com.codacy.model.configuration.CommitUUID
+import wvlet.log.LazyLogger
+
+import scala.util.Try
 
-class GitHubActionProvider extends CommitUUIDProvider {
+class GitHubActionProvider extends CommitUUIDProvider with LazyLogger {
   val name: String = "GitHub Actions"
 
-  override def validate(a: Map[String, String]): Boolean = {
-    a.get("CI").contains("true") && a.get("GITHUB_ACTIONS").contains("true")
+  override def validate(envVars: Map[String, String]): Boolean = {
+    envVars.get("CI").contains("true") && envVars.get("GITHUB_ACTIONS").contains("true")
+  }
+
+  override def getUUID(envVars: Map[String, String]): Either[String, CommitUUID] = {
+    // if the event is a pull request the GITHUB_SHA will have a different commit UUID (the id of a merge commit)
+    // https://help.github.com/en/actions/reference/events-that-trigger-workflows
+    // for this reason, we need to fetch it from the event details that GitHub provides
+    // equivalent to doing ${{github.event.pull_request.head.sha}} in a GitHub action workflow
+    if (envVars.get("GITHUB_EVENT_NAME").contains("pull_request")) {
+      getPullRequestCommit(envVars)
+    } else {
+      withErrorMessage(envVars.get("GITHUB_SHA"))
+    }
   }
 
-  override def getUUID(a: Map[String, String]): Either[String, CommitUUID] =
-    withErrorMessage(a.get("GITHUB_SHA"))
+  private def getPullRequestCommit(envVars: Map[String, String]): Either[String, CommitUUID] = {
+    for {
+      eventPath <- envVars.get("GITHUB_EVENT_PATH").toRight("Could not find event description file path")
+      eventContent <- readFile(eventPath)
+      sha <- extractHeadSHA(eventContent)
+    } yield CommitUUID(sha)
+  }
+
+  private def readFile(path: String): Either[String, String] =
+    Try(Files.readString(new File(path).toPath)).toEither.left
+      .map { t =>
+        logger.error("Failed to read event file", t)
+        s"Failed to read event file with error: ${t.getMessage}"
+      }
+
+  private def extractHeadSHA(event: String) = {
+    val eventJson = ujson.read(event)
+    Try(eventJson("pull_request")("head")("sha").str).toEither.left
+      .map(t => s"Unable to fetch SHA from event file. Failed with error: ${t.getMessage}")
+  }
```

Comment:
Here you can simplify like this:
```suggestion
    try {
      Right(eventJson("pull_request")("head")("sha").str)
    } catch {
      case NonFatal(e) => Left(s"Unable to fetch SHA from event file. Failed with error: ${t.getMessage}")
    }
```

Reply:
Using try{}catch{} will trigger an issue in Codacy. I split it so it's easier to understand though